chain/chain/src/store.rs

```diff
@@ -613,21 +606,22 @@ impl ChainStore {
     }
 
     /// Returns latest known height and time it was seen.
-    pub fn get_latest_known(&mut self) -> Result<LatestKnown, Error> {
-        if self.latest_known.is_none() {
-            self.latest_known = Some(option_to_not_found(
-                self.store.get_ser(DBCol::BlockMisc, LATEST_KNOWN_KEY),
-                "LATEST_KNOWN_KEY",
-            )?);
-        }
-        Ok(self.latest_known.as_ref().unwrap().clone())
+    pub fn get_latest_known(&self) -> Result<LatestKnown, Error> {
```

Comment:
fun bit of trivia -- in this one case, we are caching a single value. As the value is never evicted after it's cached, in this one case we actually can return a reference: 

```rust
pub fn get_latest_known(&self) -> Result<&LatestKnown, Error> {
```

But, given that LatestKnown is tiny, we clone here for consistency. 

In other words, the deep reason why we need so many arcs here is eviction -- if we returned a referencne, and then evicted the value, that would be an uaf.